File: SakhaTyla.Core/Requests/EntityChanges/Models/EntityChangeModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Text;

namespace SakhaTyla.Core.Requests.EntityChanges.Models
{
    public class EntityChangeModel
    {
        public int Id { get; set; }

        [DisplayName("Action")]
        public Enums.ChangeAction Action { get; set; }

        [DisplayName("From")]
        public string From { get; set; }

        [DisplayName("To")]
        public string To { get; set; }

        [DisplayName("Creation Date")]
        public DateTime CreationDate { get; set; }

        [DisplayName("Creation User")]
        public int? CreationUserId { get; set; }
        public Users.Models.UserShortModel CreationUser { get; set; }

        public IList<EntityPropertyChange> Changes { get; set; }
    }
}

File: SakhaTyla.Web.Ng/src/app/entity-change-core/entity-change.model.ts
import { ChangeAction } from '../change-action/change-action.model';
import { User } from '../user-core/user.model';

export class EntityChange {
    id: number;
    entityId: number;
    creationDate: Date;
    creationUserId: number;
    creationUser: User;
    action: ChangeAction;
    from: string;
    to: string;
    changes: EntityPropertyChange[];
}

export class EntityPropertyChange {
    name: string;
    displayName: string;
    fromValue: string;
    toValue: string;
}

export class EntityChangeListState {
    pageSize = 10;
    pageIndex = 0;
}
